app_integrations/batcher.py

```diff
@@ -0,0 +1,122 @@
+"""
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import json
+import math
+
+import boto3
+from botocore.exceptions import ClientError
+
+from app_integrations import LOGGER
+
+# Max lambda input payload size is 128K for the 'event' invocation type
+MAX_LAMBDA_PAYLOAD_SIZE = 128 * 1000
+
+
+class Batcher(object):
+    """Batcher class to handle segmenting large messages going to the
+    rule processor lambda function
+    """
+    LAMBDA_CLIENT = None
+
+    def __init__(self, config):
+        # Create the lambda client if it does not exist
+        if Batcher.LAMBDA_CLIENT is None:
+            Batcher.LAMBDA_CLIENT = boto3.client('lambda', region_name=config['region'])
+        # The rule processor function name will look like:
+        # <prefix>_<cluster>_streamalert_rule_processor
+        self.rp_function = '_'.join([config['prefix'], config['cluster'],
+                                     'streamalert_rule_processor'])
+
+    def send_logs(self, source_function, logs):
+        """Public method to send the logs to the rule processor
+
+        Args:
+            source_function (str): The app function name from which the logs came
+            logs (list): List of the logs that have been gathered
+        """
+        LOGGER.info('Starting batch send of %d logs to the rule processor', len(logs))
+
+        # Try to send all of the logs in one fell swoop
+        if self._send_logs_to_stream_alert(source_function, logs):
+            return
+
+        # Fall back on segmenting the list of logs into multiple requests
+        # if they could not be sent at once
+        return self._segement_and_send(source_function, logs)
+
+    def _segement_and_send(self, source_function, logs):
```

Comment:
`segment` vs. `segement` (update all other references)

Reply:
LOL it's hilarious that I typed that wrong so many times without catching this. thanks!